### DISCUSSION TITLE
Why python 3.7 for EVM?

### DISCUSSION THREAD

**Discussion by nullableVoidPtr** [qilingframework/qiling#959] [Q&A] [answered]
Hi, I noticed in setup.py:
```py
        "cached-property>=1.5.2;python_version<'3.8'",
        "typing-extensions>=3.7.4.3;python_version<'3.8'",
```
Is there any particular reason as to why EVM requires this particular version?

I didn't raise this as an issue as I am just wondering if there was something I am missing. I tried looking in CI for answers but I couldn't find any reason.

**Comment by nullableVoidPtr** [comment]
Pinging @xwings

**Reply by kabeor** [reply]
Well, these features are included in python3.8, and we use these packages for forward compatibility

**Reply by kabeor** [reply]
These packages are not required in environments greater than Python 3.8

**Comment by xwings** [comment, marked as answer]
I think its python library we rely too. feel free to test and let me know if we got it wrong.
